sklearn/utils/validation.py

```diff
@@ -947,13 +1083,19 @@ def check_array(
         if xp.__name__ in {"numpy", "numpy.array_api"}:
             # only make a copy if `array` and `array_orig` may share memory`
             if np.may_share_memory(array, array_orig):
```

Comment:
If `array` is a GPU array, I suspect `np.may_share_memory` will not work.

Reply:
But then we won't go down this path no? At least I'd assume `xp.__name__` would not be "numpy" in that case.